docs/excel/custom-functions-architecture.md

```diff
@@ -0,0 +1,42 @@
+---
+ms.date: 03/06/2019
+description: Learn about Excel custom functions' runtime. 
+title: Custom functions architecture (preview)
+localization_priority: Priority
+---
+# Custom functions architecture
+
+Custom function use a new JavaScript runtime that differs from the browser-based JavaScript engine which powers most other parts of your add-in.
+
+## Custom functions runtime
+
+The custom functions runtime prioritizes executing calculations, so functions run run smoothly when Excel recalculates. The custom functions runtime allows access to APIs which make requesting external data and exchanging data over a persistent connection over a server possible.
+
+Because functions are limited to actions performed in a workbook, the custom functions runtime doesn't support UI elements nor does it provide access to the Document Object Model (DOM). You should think of custom functions as a separate component of your add-in, separate from your add-in's task pane or other UI elements.
+
+>[!NOTE] 
```

Comment:
double-check but I don't think this is the case anymore. You always get a task pane now with the new templates.

or thinking about this more, maybe we should document the html file that is the loader for your custom functions. Just to point out its purpose and to not use it as your task pane (task panes are under the task pane folder).